go/vt/sqlparser/sql.y

```diff
@@ -856,6 +890,358 @@ default_role_opt:
+  }
+| PASSWORD HISTORY INTEGRAL
+  {
+    $$ = PassLockItem{PassLockItemType: PassLockItemType_PassHistory, Value: NewIntVal($3)}
+  }
+| PASSWORD REUSE INTERVAL DEFAULT
+  {
+    $$ = PassLockItem{PassLockItemType: PassLockItemType_PassReuseInterval, Value: nil}
+  }
+| PASSWORD REUSE INTERVAL INTEGRAL DAY
+  {
+    $$ = PassLockItem{PassLockItemType: PassLockItemType_PassReuseInterval, Value: NewIntVal($4)}
+  }
+| PASSWORD REQUIRE CURRENT DEFAULT
+  {
+    $$ = PassLockItem{PassLockItemType: PassLockItemType_PassReqCurrentDefault, Value: nil}
+  }
+| PASSWORD REQUIRE CURRENT OPTIONAL
+  {
+    $$ = PassLockItem{PassLockItemType: PassLockItemType_PassReqCurrentOptional, Value: nil}
+  }
+| FAILED_LOGIN_ATTEMPTS INTEGRAL
+  {
+    $$ = PassLockItem{PassLockItemType: PassLockItemType_PassFailedLogins, Value: NewIntVal($2)}
+  }
+| PASSWORD_LOCK_TIME INTEGRAL
+  {
+    $$ = PassLockItem{PassLockItemType: PassLockItemType_PassLockTime, Value: NewIntVal($2)}
+  }
+| PASSWORD_LOCK_TIME UNBOUNDED
+  {
+    $$ = PassLockItem{PassLockItemType: PassLockItemType_PassLockTime, Value: nil}
+  }
+| ACCOUNT LOCK
+  {
+    $$ = PassLockItem{PassLockItemType: PassLockItemType_AccountLock, Value: nil}
+  }
+| ACCOUNT UNLOCK
+  {
+    $$ = PassLockItem{PassLockItemType: PassLockItemType_AccountUnlock, Value: nil}
+  }
+
+user_comment_attribute:
+  {
+    $$ = ""
+  }
+| COMMENT_KEYWORD STRING
+  {
+    comment := string($2)
+    $$ = `{"comment": "`+escapeDoubleQuotes(comment)+`"}`
+  }
+| ATTRIBUTE STRING
+  {
+    $$ = string($2)
+  }
+
+grant_statement:
+  GRANT ALL ON grant_object_type grant_privilege_level TO account_name_list with_grant_opt grant_assumption
+  {
+    allPriv := []Privilege{Privilege{Type: PrivilegeType_All, Columns: nil}}
+    $$ = &GrantPrivilege{Privileges: allPriv, ObjectType: $4, PrivilegeLevel: $5, To: $7, WithGrantOption: $8, As: $9}
+  }
+| GRANT grant_privilege_list ON grant_object_type grant_privilege_level TO account_name_list with_grant_opt grant_assumption
+  {
+    $$ = &GrantPrivilege{Privileges: $2, ObjectType: $4, PrivilegeLevel: $5, To: $7, WithGrantOption: $8, As: $9}
+  }
+| GRANT role_name_list TO account_name_list with_admin_opt
+  {
+    $$ = &GrantRole{Roles: $2, To: $4, WithAdminOption: $5}
+  }
+| GRANT PROXY ON account_name TO account_name_list with_grant_opt
+  {
+    $$ = &GrantProxy{On: $4, To: $6, WithGrantOption: $7}
+  }
+
+revoke_statement:
+  REVOKE ALL ON grant_object_type grant_privilege_level FROM account_name_list
+  {
+    allPriv := []Privilege{Privilege{Type: PrivilegeType_All, Columns: nil}}
+    $$ = &RevokePrivilege{Privileges: allPriv, ObjectType: $4, PrivilegeLevel: $5, From: $7}
+  }
+| REVOKE grant_privilege_list ON grant_object_type grant_privilege_level FROM account_name_list
+  {
+    $$ = &RevokePrivilege{Privileges: $2, ObjectType: $4, PrivilegeLevel: $5, From: $7}
+  }
+| REVOKE ALL ',' GRANT OPTION FROM account_name_list
+  {
+    $$ = &RevokeAllPrivileges{From: $7}
```

Comment:
Thoughts on having an `All` field in `RevokePrivilege` instead? In general, better to have a single AST node encapsulate a single logical action

Reply:
Plus it looks like `Grant` already works like this

Reply:
K